daml-assistant/src/DA/Daml/Assistant/Version.hs

```diff
@@ -48,23 +56,40 @@ import qualified Data.ByteString.UTF8 as BSU
 import qualified Data.SemVer as V
 import Data.Function ((&))
 import Control.Lens (view)
+import System.Directory (listDirectory, doesFileExist)
+import System.FilePath ((</>))
+import Data.List (find)
+import Data.Either.Extra (eitherToMaybe)
 
 import qualified Data.Map.Strict as M
 import qualified Data.List.NonEmpty as NonEmpty
+import qualified System.Info
+
+import GHC.Stack
 
 -- | Determine SDK version of running daml assistant. Fails with an
 -- AssistantError exception if the version cannot be determined.
-getAssistantSdkVersion :: IO SdkVersion
-getAssistantSdkVersion = do
+getAssistantSdkVersion :: UseCache -> IO ReleaseVersion
+getAssistantSdkVersion useCache = do
     exePath <- requiredIO "Failed to determine executable path of assistant."
         getExecutablePath
     sdkPath <- required "Failed to determine SDK path of assistant." =<<
         findM hasSdkConfig (ascendants exePath)
-    getSdkVersionFromSdkPath (SdkPath sdkPath)
+    getReleaseVersionFromSdkPath useCache (SdkPath sdkPath)
     where
         hasSdkConfig :: FilePath -> IO Bool
         hasSdkConfig p = doesFileExist (p </> sdkConfigName)
 
+-- | Determine SDK version from an SDK directory. Fails with an
+-- AssistantError exception if the version cannot be determined.
+getReleaseVersionFromSdkPath :: UseCache -> SdkPath -> IO ReleaseVersion
+getReleaseVersionFromSdkPath useCache sdkPath = do
+    sdkVersion <- getSdkVersionFromSdkPath sdkPath
+    let errMsg =
+            "Failed to retrieve release version for sdk version " <> V.toText (unwrapSdkVersion sdkVersion)
+                <> " from sdk path " <> T.pack (unwrapSdkPath sdkPath)
+    requiredE errMsg =<< resolveSdkVersionToRelease useCache sdkVersion
```

Comment:
Don't we put the release version in the path? is there a reason we want to read the sdk config and check the cache over that path? Could this do less IO if we just read the path?

Reply:
It isn't necessarily in the pathname I think, and I'd rather not resort to the directory structure unless I have to.